session.go

```diff
@@ -663,14 +706,39 @@ func Connect(ctx context.Context, opts ...ConnectOption) (Session, error) {
 						if !ok {
 							return
 						}
-						cfg.HeartbeatHandler(ctx, session, latency)
+						cfg.HeartbeatHandler(ctx, heartbeatSession, latency)
 					}
 				}
 			}()
 		}
 
 		auth.Cookie = resp.Extra.Cookie
-		return nil
+
+		// store any connect server addresses for use in subsequent legs
+		if cfg.EnableMultiLeg && legNumber == 0 && len(resp.Extra.ConnectAddresses) > 1 {
+			testServerAddr := strings.Replace(cfg.ServerAddr, ".lan", "", 1)
+			testServerAddr = strings.Replace(testServerAddr, ".dev-", ".", 1)
+			testServerAddr = strings.Replace(testServerAddr, ".stage-", ".", 1)
```

Comment:
test code?

Reply:
i meant "test" there as in testing equality with default hostname, but it does bake some of our testing environment naming in here, which isn't the best. it needs to know if this is a least-latency default address vs a customer-specified one, and we set the server_addr for those environments concretely. maybe it should be generalized to a regex.

Reply:
I switched this to a regex to avoid the replace awkwardness.